### PR TITLE
klong: init at 20221212

### PR DESCRIPTION
**Pull request by casaca24**
## Description of changes
Klong is a programming language that uses the Array programming paradigm.

ex: generate a list of numbers (0..4) and then add(+) 1 to each element
    1+!5 = [1 2 3 4 5]

no build deps besides gcc and make

https://t3x.org/klong

- Built on platform(s)
  - [x] x86_64-linux

### CODE REVIEW

File: pkgs/development/interpreters/klong/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper }:
+
+
+
+stdenv.mkDerivation rec {
+  name = "klong";
```
Comment (Aleksanaa) [inline pkgs/development/interpreters/klong/default.nix:9]:
```suggestion
  pname = "klong";
```

File: pkgs/development/interpreters/klong/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  version = "20221212";
+
+  src = fetchurl {
+    url = "https://t3x.org/klong/${name}${version}.tgz";
```
Comment (Aleksanaa) [inline pkgs/development/interpreters/klong/default.nix:13]:
```suggestion
    url = "https://t3x.org/klong/klong${version}.tgz";
```

File: pkgs/development/interpreters/klong/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    sha256 = "sha256-2IuD3sanS4Rmq0CnBqWTJWx7SJdz2EMdn+8uKhBAC7c=";
```
Comment (Aleksanaa) [inline pkgs/development/interpreters/klong/default.nix:14]:
```suggestion
    hash = "sha256-2IuD3sanS4Rmq0CnBqWTJWx7SJdz2EMdn+8uKhBAC7c=";
```

File: pkgs/development/interpreters/klong/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    install -m 555 -Dt $out/bin kg kplot
+  '';
```
Comment (Aleksanaa) [inline pkgs/development/interpreters/klong/default.nix:22]:
```suggestion
  installPhase = ''
    runHook preInstall
    install -m 555 -Dt $out/bin kg kplot
    runHook postInstall
  '';
```

File: pkgs/development/interpreters/klong/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  fixupPhase = ''
+    mkdir -p $out/lib
+    cp lib/*.kg $out/lib
+    wrapProgram $out/bin/kg --prefix KLONGPATH $out/lib
+  '';
```
Comment (Aleksanaa) [inline pkgs/development/interpreters/klong/default.nix:28]:
```suggestion
  postFixup = ''
    mkdir -p $out/lib/klong
    cp lib/*.kg $out/lib/klong
    wrapProgram $out/bin/kg --prefix KLONGPATH $out/lib/klong
  '';
```

File: pkgs/development/interpreters/klong/default.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/development/interpreters/klong/default.nix:30]:
```suggestion
  meta = {
```

Add `lib` below.

File: pkgs/development/interpreters/klong/default.nix [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    description = "A simple Array programming language.";
```
Comment (Aleksanaa) [inline pkgs/development/interpreters/klong/default.nix:31]:
```suggestion
    description = "Simple Array programming language";
```

File: pkgs/development/interpreters/klong/default.nix [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    homepage = "https://t3x.org/klong";
+    license = licenses.publicDomain;
+    mainProgram = "kg";
+    platforms = platforms.all;
```
Comment (Aleksanaa) [inline pkgs/development/interpreters/klong/default.nix:35]: add your self to maintainers/maintainer-list.nix, and maintainer entry here

**Review by Aleksanaa — COMMENTED** [review]
Please format with `nixfmt-rfc-style`, and squash commit into:

```
maintainers: add casaca
klong: init at 20221212
```

**Review by vinnymeller — COMMENTED** [review]
[Here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#category-hierarchy) is the README for the package category hierarchy

I think package should go in the `pkgs/by-name` folder instead of where it is, so instead of the current location, it would be  `pkgs/by-name/kl/klong/package.nix`

Also for the future, Aleksanaa's requested changes for the commits come from these sections: [Commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) and [Commit conventions for `pkgs`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions)

File: pkgs/development/interpreters/klong/default.nix (hunk 8 above)
Comment (casaca24) [reply pkgs/development/interpreters/klong/default.nix:35]: thank you very much!

File: pkgs/development/interpreters/klong/default.nix (hunk 6 above)
Comment (casaca24) [reply pkgs/development/interpreters/klong/default.nix:30]: i'm having trouble with this one, where do add `lib` again?

Comment (casaca24) [reply pkgs/development/interpreters/klong/default.nix:30]: got it, thank you.

**Comment by donovanglover** [comment]
Make sure to change your git user settings if you want to associate your commits with your GitHub account

File: pkgs/by-name/kl/klong/package.nix [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  installPhase = ''
+    runHook preInstall
+    install -m 555 -Dt $out/bin kg kplot
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    mkdir -p $out/lib/klong
+    cp lib/*.kg $out/lib/klong
+    wrapProgram $out/bin/kg --prefix KLONGPATH $out/lib/klong
```
Comment (Aleksanaa) [inline pkgs/by-name/kl/klong/package.nix:28]:
```suggestion
    wrapProgram $out/bin/kg --prefix KLONGPATH : $out/lib/klong
```

I suppose?

File: pkgs/development/interpreters/klong/default.nix (hunk 7 above)
Comment (Aleksanaa) [reply pkgs/development/interpreters/klong/default.nix]: This is not fixed

**Review by Aleksanaa — APPROVED** [review]

**Review by vinnymeller — APPROVED** [review]

**Review by vinnymeller — CHANGES_REQUESTED** [review]
just kidding:

Result of `nixpkgs-review pr 341279` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>klong</li>
  </ul>
</details>

Looks like the hash is wrong

```
error: hash mismatch in fixed-output derivation '/nix/store/ck3gc6s14zq8p9cmpwvvbdk8jvcmyhl8-klong20221212.tgz.drv':
         specified: sha256-2IuD3sanS4Rmq0CnBqWTJWx7SJdz2EMdn+8uKhBAC7c=
            got:    sha256-XhpIdyKKPGQ6mdv9LXPmC8P6hW4mFawv54yANw5/lrQ=
```

**Comment by casaca24** [comment]
> just kidding:
> 
> Result of `nixpkgs-review pr 341279` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 1 package failed to build:
> Looks like the hash is wrong
> 
> ```
> error: hash mismatch in fixed-output derivation '/nix/store/ck3gc6s14zq8p9cmpwvvbdk8jvcmyhl8-klong20221212.tgz.drv':
>          specified: sha256-2IuD3sanS4Rmq0CnBqWTJWx7SJdz2EMdn+8uKhBAC7c=
>             got:    sha256-XhpIdyKKPGQ6mdv9LXPmC8P6hW4mFawv54yANw5/lrQ=
> ```

thank you all for your patience.

**Review by vinnymeller — APPROVED** [review]
Result of `nixpkgs-review pr 341279` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>klong</li>
  </ul>
</details>

File: pkgs/development/interpreters/klong/default.nix (hunk 7 above)
Comment (Aleksanaa) [reply pkgs/development/interpreters/klong/default.nix]: This is still not fixed

**Comment by casaca24** [comment]
apologies, I incorrectly updated my fork, and messed it up. This fixes it

**Comment by casaca24** [comment]
Added a manual page, and removed the trailing dot from `description` in the meta section.

**Comment by casaca24** [comment]
sorry for the noise, i think my force foce pushes and rebasing changes are conflicting with parts of the nixpkgs build. should i close this PR and open another? 

no hurry :)

File: pkgs/by-name/kl/klong/package.nix [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  version = "20221212";
+  src = fetchurl {
+    url = "https://t3x.org/klong/klong${version}.tgz";
+    hash = "sha256-XhpIdyKKPGQ6mdv9LXPmC8P6hW4mFawv54yANw5/lrQ=";
+  };
+
+  docs = fetchFromSourcehut {
+    owner = "~nut";
+    repo = "klong-docs";
+    hash = "sha256-ASrkk7bcrXB3WT4rkSKOl4nop2UtQXekJ0pCgIpQEjU=";
```
Comment (Aleksanaa) [inline pkgs/by-name/kl/klong/package.nix:21]: you are missing `rev` here

Comment (casaca24) [reply pkgs/by-name/kl/klong/package.nix:21]: thanks! ill fix it

File: maintainers/maintainer-list.nix [hunk 11]
```diff
@@ -3422,6 +3422,10 @@
     githubId = 5241813;
     name = "Carter Sande";
   };
+  casaca = {
+    name = "J McNutt";
+    email = "jmacasac@hotmail.com";
```
Comment (Aleksanaa) [inline maintainers/maintainer-list.nix:3427]: This won't work nowadays. You won't get pinged when there's someone opening issues or PRs

**Comment by gshpychka** [comment]
Result of `nixpkgs-review pr 341279` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>klong</li>
  </ul>
</details>

File: pkgs/by-name/kl/klong/package.nix [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    installManPage $docs/*.1
+    mkdir -p $out/lib/klong
+    cp lib/*.kg $out/lib/klong
+    wrapProgram $out/bin/kg --prefix KLONGPATH : $out/lib/klong
+  '';
+
+  meta = {
+    description = "Simple Array programming language";
+    homepage = "https://t3x.org/klong";
+    license = lib.licenses.publicDomain;
```
Comment (thecaralice) [inline pkgs/by-name/kl/klong/package.nix:47]:
```suggestion
    license = with lib.licenses; [ publicDomain cc0 ];
```
klong allows both public domain and CC0.

> Nobody owns this code.
> It's in the public domain or whatever you call it.
> 
> If your jurisdiction does not offer the concept of the
> public domain, the Creative Commons Zero (CC0) license applies.
> See the file CC0 for a copy of the CC0 license.

**Review by magistau — CHANGES_REQUESTED** [review]

**Comment by thecaralice** [comment]
Result of `nixpkgs-review pr 341279` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>klong</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix (hunk 11 above)
Comment (casaca24) [reply maintainers/maintainer-list.nix:3427]: OK, on it.

File: pkgs/by-name/kl/klong/package.nix [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    installShellFiles
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    install -m 555 -Dt $out/bin kg kplot
```
[the remark this answers is not in the document]
Comment (casaca24) [reply pkgs/by-name/kl/klong/package.nix:33]: thanks for the input, ill make it work

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**